velox/exec/HashPartitionFunction.cpp

```diff
@@ -57,7 +57,12 @@ void HashPartitionFunction::init(
 std::optional<uint32_t> HashPartitionFunction::partition(
     const RowVector& input,
     std::vector<uint32_t>& partitions) {
-  auto size = input.size();
+  const auto size = input.size();
+
+  if (hashers_.empty() && !hashBitRange_.has_value()) {
```

Comment:
@usurai This logic is surprising. Let's document it in the class-level comments in the .h file. Let's also return non-empty optional instead of filling in partitions with zeros. 

```
return {0};
```

Reply:
I've reviewed the usage of partition(), and it appears that certain code, such as in HashProbe and SpillerTest, doesn't check the return value. I suggest either ensuring that partitions are properly filled with zero or updating the code in those areas to check the return value.

Reply:
> updating the code in those areas to check the return value.

Let's do that.

Reply:
I created a separate PR to address this: https://github.com/facebookincubator/velox/pull/6534

Reply:
Thanks.